DevEdu.API/DevEdu.Business/Services/TaskService.cs

```diff
@@ -55,11 +79,16 @@ public int AddTask(TaskDto taskDto)
 
         public TaskDto UpdateTask(TaskDto taskDto)
         {
+            _taskValidationHelper.CheckTaskExistence(taskDto.Id);
```

Comment:
it's a mistake that you are passing id of the task inside the dto
please take a look at any other update method: id is passed as a parameter
so the line "taskDto.Id = taskId;" must be moved from the controller into this method